opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1263,8 +1281,16 @@ namespace Opm {
             std::vector<int> failed_cells_pb;
             std::vector<int> failed_cells_pd;
 
+            if ( ebosModel.cachedIntensiveQuantities(/*cellIdx=*/0, /*timeIdx=*/0) == nullptr ) {
+                updateSimCache();
+            }
+
             for (int cellIdx = 0; cellIdx < numCells; ++cellIdx) {
```

Comment:
also, this code may cause trouble in the parallel case because the grid is distributed at this point. I'm not sure about this, though.